pkg/aws/eni/routing/routing.go

```diff
@@ -62,71 +63,73 @@ type RoutingInfo struct {
 // mtu: The ENI device MTU.
 // masq: Whether masquerading is enabled.
 func Install(ip net.IP, info *RoutingInfo, mtu int, masq bool) error {
-	ifindex, err := retrieveIfIndexFromMAC(info.MasterIfMAC, mtu)
-	if err != nil {
-		return fmt.Errorf("unable to find ifindex for interface MAC: %s", err)
-	}
 
-	ipWithMask := net.IPNet{
-		IP:   ip,
-		Mask: net.CIDRMask(32, 32),
-	}
+	if (option.Config.IPAM == option.IPAMENI) && (ip.To4() != nil) {
```

Comment:
if you invert this if statement this function can return earlier and we can avoid touching the remaining code of this function.

Reply:
agreed